### ISSUE TITLE
Say more about InternalErrors

### ISSUE THREAD

**Issue by michaelpj** [nixbuild/feedback#22]
Something seems bad here but I have no idea what!

```
1264885 2023-03-23 14:55:28Z 12.00 s 1 CPU [Running]
        /nix/store/8dk66cn3rwqcs03ayjjjww1gay28rvgd-orphans-deriving-via-lib-orphans-deriving-via-0.1.0.1.drv
1264875 2023-03-23 14:51:42Z 03:37 1 CPU [InternalError]
        /nix/store/8dk66cn3rwqcs03ayjjjww1gay28rvgd-orphans-deriving-via-lib-orphans-deriving-via-0.1.0.1.drv
1264874 2023-03-23 14:51:32Z 4.00 s 1 CPU [InternalError]
        /nix/store/8dk66cn3rwqcs03ayjjjww1gay28rvgd-orphans-deriving-via-lib-orphans-deriving-via-0.1.0.1.drv
```

**Comment by michaelpj** [comment]
It succeeded on the 5th try! :o

**Comment by rickynils** [comment]
I don't know about this particular case (5 tries sounds excessive, maybe this was caused by a bug or lockup in nixbuild.net), but some "internal errors" are expected from time to time. This is because some of our builder machines are spot-instances that can be replaced in the middle of a build.  Sometimes hardware errors on builder machines can also cause builds to fail. This is something nixbuild.net is built to handle by simply restarting the build. Users are never charged for the builds that restarted.

Should we replace the term "internal error" with something that better communicates the above? Do you have a suggestion, perhaps?

**Comment by michaelpj** [comment]
I think the issue is mainly that I don't know how to interpret this. Perhaps the issue is really https://github.com/nixbuild/feedback/issues/23. Mostly I don't know if seeing `InternalError` there is action-relevant for me or not. Is it something I need to investigate? Raise here? Who knows! 

Basically `InternalError` could mean:
1. Uh oh, something bad that should never happen has happened! Pull the fire alarm!
2. Nothing to see here, just some internal stuff that happens occasionally.

It sounds like the situation is 2, but that's not obvious to the user atm.

**Comment by rickynils** [comment]
Yeah, it is definitely situation 2. I'll think about how to communicate that better.
